### ISSUE TITLE
Windows ECS Agent breaks on settings that take values from boolean environment variables

### ISSUE THREAD

**Issue by nonspecialist** [aws/amazon-ecs-agent#2571]
### Summary
Windows agents where `parseBooleanDefault...` are used fail since https://github.com/aws/amazon-ecs-agent/commit/481a2487c86d9ef6f9a0b3f92a9be7a886db1767#diff-b4442ae5495d1d454858bd0fc134435e 

### Description
When environment variables are set using the form `[Environment]::SetEnvironmentVariable("ECS_ENABLE_AWSLOGS_EXECUTIONROLE_OVERRIDE", $TRUE, "Machine")` -- previously, the settings were parsed into `true` or `false` internally, but now the implementation appears to require a _string_ representation of the values, instead of the Windows internal boolean type. This has broken clusters that were built with agent 1.43.0, and possibly with 1.42.0 as well since no tasks can be scheduled (these workers are missing a required capability)

### Expected Behavior
Using previous configuration (with `$TRUE`) would still work since no breaking change was announced

### Observed Behavior
Services cannot schedule tasks on new instances with the error 

```
service banana_foo was unable to place a task because no container instance met all of its requirements. The closest matching container-instance cc293386c3e34ea48de8e561efae6583 is missing an attribute required by your task
```

### Environment Details
AWS-supplied Windows Server 1909 ECS instances with agent 1.43.0

### Supporting Log Snippets
```
level=warn time=2020-08-17T06:48:20Z msg="Invalid format for \"ECS_ENABLE_AWSLOGS_EXECUTIONROLE_OVERRIDE\", expected a boolean. err invalid character 'T' looking for beginning of value" module=parse.go
```

### Workaround
Change `UserData` or other configuration mechanisms to use the string `true` instead of the Windows environment variable boolean type `$True` or `$TRUE`

**Comment by 4wuyan** [comment]
Also in https://github.com/aws/amazon-ecs-agent/commit/211e2b#diff-82751e0f6acee8ce7c2f62cfe46735ad
some settings used to be parsed via `utils.ParseBool`, which uses `strconv.ParseBool`, which according to [doc](https://golang.org/pkg/strconv/#ParseBool)
> It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.

But now they are parsed via `parseBooleanDefaultFalseConfig`, which uses `json.Unmarshal`, which I believe only accepts `"true"` or `"false"`.

Although those environment variables should have the value of `"true"` or `"false"` when Windows users are setting them, it's tempting in Powershell to do
```powershell
$env:x = $true
```
which powershell automatically converts to and stores as string `"True"` in `env:`. Hence it could break some existing code.

**Comment by ubhattacharjya** [comment]
Hi,

This fix for this issue has been fixed in https://github.com/aws/amazon-ecs-agent/pull/2573  and the change has been released with agent version 1.44.1
Closing this issue.

Kindly reopen if you still face the same problem
